extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/health/MongoHealthCheck.java

```diff
@@ -17,40 +16,78 @@
 import com.mongodb.client.MongoClient;
 
 import io.quarkus.arc.Arc;
+import io.quarkus.mongodb.MongoClientName;
+import io.quarkus.mongodb.reactive.ReactiveMongoClient;
 
 @Readiness
 @ApplicationScoped
 public class MongoHealthCheck implements HealthCheck {
-    private static final String DEFAULT_CLIENT = "__default__";
+
     private Map<String, MongoClient> clients = new HashMap<>();
+    private Map<String, ReactiveMongoClient> reactiveClients = new HashMap<>();
 
     @PostConstruct
     protected void init() {
-        Set<Bean<?>> beans = Arc.container().beanManager().getBeans(MongoClient.class);
-        for (Bean<?> bean : beans) {
+        for (Bean<?> bean : Arc.container().beanManager().getBeans(MongoClient.class)) {
+            // Get client name from annotation because InjectableBean.getName() returns always null.
+            String clientName = getMongoClientName(bean);
+            MongoClient client;
             if (bean.getName() == null) {
                 // this is the default mongo client: retrieve it by type
-                MongoClient defaultClient = Arc.container().instance(MongoClient.class).get();
-                clients.put(DEFAULT_CLIENT, defaultClient);
+                client = Arc.container().instance(MongoClient.class).get();
             } else {
-                MongoClient client = (MongoClient) Arc.container().instance(bean.getName()).get();
-                clients.put(bean.getName(), client);
+                client = (MongoClient) Arc.container().instance(bean.getName()).get();
+            }
+            clients.put(clientName == null ? "default" : clientName, client);
```

Comment:
In other parts of the code, we use `<default>` so I would go with `<default>` and `<default-reactive>`.

Also let's make them public constants that you can use in the tests. It's more practical and doesn't cost much.

Reply:
This would be a small breaking change in case some people parsed the response and search for `__default__` but for cross extension consistancy I'm OK with it.

Reply:
sure thing. Will change it after first question answer.